src/charm.py

```diff
@@ -0,0 +1,1624 @@
+#!/usr/bin/env python3
```

Comment:
I strongly advise to split this file into multiple modules

Reply:
We plan on doing that during the next cycle :)